nomic/dataset.py

```diff
@@ -156,10 +161,74 @@ def _get_project_by_id(self, project_id: str):
         )
 
         if response.status_code != 200:
-            raise Exception(f"Could not access project with id {project_id}: {response.text}")
+            raise Exception(f"Could not access dataset with id {project_id}: {response.text}")
+
+        return response.json()
+
+    def _get_organization_by_slug(self, slug: str):
+        '''
+
+        Args:
+            slug: The organization slug
+
+        Returns:
+            Returns the requested project.
+        '''
+
+        if '/' in slug:
+            slug = slug.split('/')[0]
+
+        response = requests.get(
+            self.atlas_api_path + f"/v1/organization/search/{slug}",
+            headers=self.header,
+        )
+
+        if response.status_code != 200:
+            raise Exception(f"Organization not found: {slug}")
+
+        return response.json()['organization_id']
+
+    def _get_project_by_slug_identifier(self, identifier: str):
+        '''
+
+        Args:
+            identifier: the organization slug and dataset slug seperated by a slash
+
+        Returns:
+            Returns the requested project.
+        '''
+
+        if not self.is_valid_dataset_identifier(identifier=identifier):
+            raise Exception("Invalid dataset identifier")
+
+        organization_slug = identifier.split('/')[0]
+        project_slug = identifier.split('/')[1]
+        response = requests.get(
+            self.atlas_api_path + f"/v1/project/{organization_slug}/project/{project_slug}",
+            headers=self.header,
+        )
```

Comment:
I want to get rid of that `/project/` route in favor of the shorter format.
```suggestion
        response = requests.get(
            self.atlas_api_path + f"/v1/project/{identifier}",
            headers=self.header,
        )
```

Reply:
yeah that didn't exist when i wrote this code :)